feat(fx): Improve Fx exp features page (#39765)
mdn/content: files/en-us/mozilla/firefox/experimental_features/index.md

{{FirefoxSidebar}}

This page lists Firefox's experimental and partially-implemented features, including evolving or proposed web platform standards.

> Each entry below includes information on the builds in which the features are included, whether they are activated by default or not, and which **preference** can be used to activate or deactivate them.
> The description of each feature also includes links to relevant [bugzilla bugs](https://bugzilla.mozilla.org) which implement or enable the feature.
> This allows you to test features before they are released for feedback and experimentation.

Reviewer: ```suggestion
Each entry below includes information about the builds–Nightly, Developer Edition, Beta, or Release–in which a feature is included, whether it is activated by default or not, and the name of the **preference** you can use to activate (set to `true`) or deactivate (set to `false`) the feature.
The description of each feature also includes links to relevant [Bugzilla bugs](https://bugzilla.mozilla.org) that implement or enable the feature.
This information allows you to try out experimental features and provide feedback before they are officially released.
```
OR
```suggestion
Each entry below includes:
- The builds–Nightly, Developer Edition, Beta, or Release–in which a feature is included
- Whether the feature is activated by default
- Name of the **preference** you can use to activate (set to `true`) or deactivate (set to `false`) the feature
- Links to relevant [Bugzilla bugs](https://bugzilla.mozilla.org) that implement or enable the feature.

With this information, you can try out experimental features and provide feedback before they are officially released.
```

Author: I like the first one, but small formatting change

> In terms of lifecycle, new features usually appear first in [Firefox Nightly](https://www.mozilla.org/en-US/firefox/channel/desktop/), where they are often enabled by default.
> If there's no major problems detected, the changes then land in [Firefox Developer Edition](https://www.mozilla.org/en-US/firefox/developer/), before finally being shipped in the stable [release version](https://www.mozilla.org/en-US/firefox/new/).
> When a feature is enabled by default in a release, it is no longer considered experimental and is removed from this page.

Reviewer: Should we also include beta? 
```suggestion
In terms of lifecycle, new features usually appear first in [Nightly](https://www.mozilla.org/en-US/firefox/channel/desktop/#nightly), where they are often enabled by default and undergo early testing.
If no major problems are found, they move to [Beta](https://www.mozilla.org/en-US/firefox/channel/desktop/#beta), which is used for broader testing and also forms the basis for [Developer Edition](https://www.mozilla.org/en-US/firefox/developer/). Finally, tested and approved features ship in the stable [Release](https://www.mozilla.org/en-US/firefox/new/) channel.
When a feature is enabled by default in a release, it is no longer considered experimental and is removed from this page.
```

> To enable these features, change the associated **preference** at `about:config` in the Firefox address bar.
> Depending on the feature, restarting the browser might be necessary for the change in preferences to be successfully applied.

Reviewer: - The previous verbiage was mostly fine. I think saying "**preference** at `about:config`" might mean that the preferences are stored there, whereas `about:config` is just the UI to view or change the preferences.
- Also added another step to search for the preference.
```suggestion
To enable these features, enter `about:config` in the Firefox address bar, search for the associated **preference** and toggle its value between `true` and `false`.
Depending on the feature, you might need to restart the browser for the change to take effect.
```

Check the [Firefox Configuration Editor](https://support.mozilla.org/en-US/kb/about-config-editor-firefox) support article for more information about managing preferences in Firefox.

## HTML

### Layout for input type="search"

Layout for `input type="search"` has been updated. This causes a search field to have a clear icon once someone starts typing in it, to match other browser implementations. (See [Firefox bug 558594](https://bugzil.la/558594) for more details.)

- `layout.forms.input-type-search.enabled`

>   - : Set to `true` to enable.

Reviewer: Oh! Was your intention in using a DL to get an anchor link for the pref name?

>   - : Set to `true` to enable.

Author: > Was your intention in using a DL to get an anchor link for the pref name

Yeah this is one bonus from DLs, although I also like your suggestion.

>   - : Set to `true` to enable.

Author: see e61570ba3b7ae581755e9836269a3fce4e50147f

- `-moz-transform-origin`
- `-moz-transform-style`

- `layout.css.prefixes.transforms`

>   - : Set to `true` to enable.

Author: fixed in e61570ba3b7ae581755e9836269a3fce4e50147f

- `-moz-transform-origin`
- `-moz-transform-style`

| Release channel   | Version added | Enabled by default? |
| ----------------- | ------------- | ------------------- |
| Nightly           | 120           | Yes                 |
| Developer Edition | 120           | Yes                 |
| Beta              | 120           | Yes                 |
| Release           | 120           | Yes                 |

- `layout.css.prefixes.transforms`
  - : Set to `true` to enable.

### UA styles for `<h1>` nested in sectioning elements

The `<h1>` heading doesn't decrease in font size now when nested within [sectioning elements](/en-US/docs/Web/HTML/Guides/Content_categories#sectioning_content) `<article>`, `<aside>`, `<nav>`, and `<section>`. The UA styles for `<h1>` nested within sectioning elements are no longer relevant since the outline algorithm [has been removed](https://github.com/whatwg/html/pull/7829) from the HTML specification. ([Firefox bug 1883896](https://bugzil.la/1883896)).

> [!NOTE]

> > The preference for this feature works in reverse: it's set to `false` in the Nightly build, which removes the UA styling for headings nested in sectioning elements.
> > It's set to `true` in all other channels, which retains the existing UA styling for the nested headings.

Reviewer 2: ```suggestion
> The preference for this feature works in reverse: when set to `false`, it removes the UA styling for headings nested in sectioning elements.
> When set to `true`, it retains the existing UA styling for the nested headings.
```

This feature is now partially enabled in other channels as well. So I wouldn’t mention it in the description and keep this info in the table.

Author: nice, tnx

- [`CSSPositionTryRule`](/en-US/docs/Web/API/CSSPositionTryRule) and [`CSSPositionTryDescriptors`](/en-US/docs/Web/API/CSSPositionTryDescriptors) (Firefox 131).

- `layout.css.anchor-positioning.enabled`

>   - : Set to `true` to enable.

Author: fixed in e61570ba3b7ae581755e9836269a3fce4e50147f

- `islamic-rgsa`
- `islamic-umalqura`

- `javascript.options.experimental.temporal`

>   - : Set to `true` to enable.

Author: fixed in e61570ba3b7ae581755e9836269a3fce4e50147f

- The global [`trustedTypes`](/en-US/docs/Web/API/Window/trustedTypes) property is available for accessing the Trusted Types API.
- The properties {{domxref("Element.innerHTML")}} and {{domxref("ShadowRoot.innerHTML")}} can be called with [trusted types](/en-US/docs/Web/API/Trusted_Types_API).

- `dom.security.trusted_types.enabled`

>   - : Set to `true` to enable.

Author: fixed in e61570ba3b7ae581755e9836269a3fce4e50147f

| ----------------- | ------------- | ------------------- |
| Nightly           | 59            | No                  |
| Developer Edition | 59            | No                  |
| Beta              | 59            | No                  |
| Release           | 59            | No                  |

- `security.mixed_content.block_object_subrequest`
  - : Set to `true` to enable.

### Insecure page labeling

The two `security.insecure_connection_text_*` preferences add a "Not secure" text label in the address bar next to the traditional lock icon when a page is loaded insecurely (that is, using {{Glossary("HTTP")}} rather than {{Glossary("HTTPS")}}). The `browser.urlbar.trimHttps` preference trims the `https:` prefix from address bar URLs. See [Firefox bug 1853418](https://bugzil.la/1853418) for more details.

| Release channel   | Version added | Enabled by default? |
| ----------------- | ------------- | ------------------- |
| Nightly           | 121           | Yes                 |
| Developer Edition | 60            | No                  |
| Beta              | 60            | No                  |
| Release           | 60            | No                  |

> - `security.insecure_connection_text.pbmode.enabled`
>   - : Set to `true` to enable the text label for private browsing mode.
> - `browser.urlbar.trimHttps`
>   - : Set to `true` to trim the `https:` prefix from address bar URLs.
> - `security.insecure_connection_text.enabled`
>   - : Set to `true` to enable the text label for normal browsing mode.

Reviewer 2: ```suggestion
- `security.insecure_connection_text.enabled`
  - : Set to `true` to enable the text label for normal browsing mode.
- `security.insecure_connection_text.pbmode.enabled`
  - : Set to `true` to enable the text label for private browsing mode.
- `browser.urlbar.trimHttps`
  - : Set to `true` to trim the `https:` prefix from address bar URLs.
```

I’d keep the same order as in the original table

Author: good idea

- `security.mixed_content.block_object_subrequest`
  - : Set to `true` to enable.

| Release channel   | Version added | Enabled by default? |
| ----------------- | ------------- | ------------------- |
| Nightly           | 59            | No                  |
| Developer Edition | 59            | No                  |
| Beta              | 59            | No                  |
| Release           | 59            | No                  |

### Insecure page labeling

The two `security.insecure_connection_text_*` preferences add a "Not secure" text label in the address bar next to the traditional lock icon when a page is loaded insecurely (that is, using {{Glossary("HTTP")}} rather than {{Glossary("HTTPS")}}). The `browser.urlbar.trimHttps` preference trims the `https:` prefix from address bar URLs. See [Firefox bug 1853418](https://bugzil.la/1853418) for more details.

- `browser.urlbar.trimHttps`
  - : Set to `true` to trim the `https:` prefix from address bar URLs.
- `security.insecure_connection_text.enabled`
  - : Set to `true` to enable the text label for normal browsing mode.
- `security.insecure_connection_text.pbmode.enabled`

>   - : Set to `true` to enable the text label for private browsing mode.

Author: Resolving for now after e61570ba3b7ae581755e9836269a3fce4e50147f

Set the preference to `false` to ensure Firefox does not include the `Authorization` header.
For more details see [Firefox bug 1687364](https://bugzil.la/1687364).

- `network.cors_preflight.authorization_covered_by_wildcard`
  - : Set to `true` to enable.

| Release channel   | Version added | Enabled by default? |
| ----------------- | ------------- | ------------------- |
| Nightly           | 115           | Yes                 |
| Developer Edition | 115           | Yes                 |
| Beta              | 115           | Yes                 |
| Release           | 115           | Yes                 |

> ## Developer tools

Author: Removing might break links to the anchor, so I've just added a small note in e61570ba3b7ae581755e9836269a3fce4e50147f
